code/go/0chain.net/smartcontract/zcnsc/authorizer_test.go

```diff
@@ -101,7 +101,7 @@ func TestAuthorizerNodeShouldBeAbleToAddTransfer(t *testing.T) {
 	// Response params
 	responseDigPoolEqualityCheck(t, resp, tr, &an.Staking.ID, &stringEmpty)
 
-	err = sc.AddTransfer(transfer)
+	err = sc.AddTransfer(transfer) //nolint: typecheck
```

Comment:
Not sure exactly why its complaining, my guess its confused by this bit of code.
```go
	var transfer *state.Transfer
	transfer, resp, err := an.Staking.DigPool(tr.Hash, tr)
```
Maybe try being consistent here either remove `var transfer *state.Transfer` or maybe add `var resp string; var err error` and use `=` instead of `:=`